ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetPayloadV2Test.java

```diff
@@ -63,6 +80,70 @@ public void shouldReturnBlockForKnownPayloadId() {
     verify(engineCallListener, times(1)).executionEngineCalled();
   }
 
+  @Test
+  public void shouldReturnBlockWithCorrectValue() {
```

Comment:
Would prefer if this was tested at the BlockResultFactory level, is there a reason you've done it here?

Reply:
Updated